src/ctim/schemas/common.clj

```diff
@@ -37,6 +31,54 @@
 
 (def default-tlp "green")
 
+(s/defschema BaseEntity
+  {;; :id and :idref must be implemented exclusively
+   :id ID
+   :type s/Str
+   :schema_version (describe s/Str "CTIM schema version for this entity")
+   (s/optional-key :uri) URI
+   (s/optional-key :revision) s/Int
+   (s/optional-key :external_ids) [s/Str]
+   (s/optional-key :timestamp) Time
+   (s/optional-key :language) s/Str
+   (s/optional-key :tlp) TLP
+   })
+
+(s/defschema NewBaseEntity
+  "Base for New Entities, optionalizes ID and type and schema_version"
+  (st/merge
+   BaseEntity
+   (st/optional-keys
+    {:id ID
+     :type (describe s/Str "A valid entity type identifer")
+     :schema_version (describe s/Str "CTIM schema version for this entity")})))
```

Comment:
I would make the accepted value to be an enum of the current schema version, this way the client would have to provide a version string matching the one currently used and fail otherwise.